integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/tests/ServletApiTestApp.java

```diff
@@ -75,6 +78,58 @@ private void testCaptureBody(AbstractServletContainerIntegrationTest test) throw
         }
     }
 
+    private void testTransactionReportingWithForward(AbstractServletContainerIntegrationTest test) throws Exception {
+        String pathToTest = "/simple-webapp" + "/forward";
+        boolean isExistForwardSpan = false;
+        boolean isExistDbH2QuerySpan = false;
+        test.clearMockServerLog();
+
+        test.executeAndValidateRequest(pathToTest, "Hello World", 200, null);
+
+        JsonNode transaction = test.assertTransactionReported(pathToTest, 200);
+
+        List<JsonNode> reportedSpans = test.getReportedSpans();
+        assertThat(reportedSpans.size()).isEqualTo(2);
+
+        for (JsonNode span : test.getReportedSpans()) {
+            String spanType = span.get("type").textValue();
+            if ("servlet.request-dispatcher.forward".equals(spanType)) {
+                isExistForwardSpan = true;
+                assertThat(span.get("name").textValue()).isEqualTo("FORWARD /servlet");
+            } else if ("db.h2.query".equals(spanType)) {
+                isExistDbH2QuerySpan = true;
+            }
+        }
```

Comment:
You need to assert that exactly the expected spans are captured. Potentially, this can pass if two non-related spans are captured.
You can `filter` the span collection for each and assert you get exactly one that contains the required info.

Reply:
added filter before assert.